core/tasks/handler/contact_tasks.go

```diff
@@ -160,7 +160,12 @@ func HandleChannelEvent(ctx context.Context, rt *runtime.Runtime, eventType mode
 	}
 
 	if models.ContactSeenEvents[eventType] {
-		err = modelContact.UpdateLastSeenOn(ctx, rt.DB, event.CreatedOn())
+		lastSeenOn := event.CreatedOn()
+		if lastSeenOn.IsZero() {
+			lastSeenOn = time.Now()
+		}
```

Comment:
I don't think we want to ever should use `event.CreatedOn`... 

Just call `UpdateLastSeenOn` with `dates.Now()`